### PR TITLE
Calculator: Added keyboard input

### PR DESCRIPTION
**Pull request by rhin123**

### CODE REVIEW

File: Applications/Calculator/CalculatorWidget.cpp [hunk 1]
```diff
@@ -197,3 +197,46 @@ void CalculatorWidget::update_display()
...
+        break;
+    case KeyCode::Key_Return:
+        clicked_button = m_equals_button;
+        break;
+    default:
+        break;
+    }
+
+    //Check for operand keys
+    if (event.key() > 51 && event.key() < 62) {
```
Comment (awesomekling) [inline Applications/Calculator/CalculatorWidget.cpp]: Let's not rely on the integer values of the `KeyCode` enum here.

**Comment by rhin123** [comment]
Good idea, fixed 😄

**Comment by bugaevc** [comment]
`#include "usual_complaint_about_commit_message.h"`

File: Applications/Calculator/CalculatorWidget.cpp [hunk 2]
```diff
@@ -197,3 +197,46 @@ void CalculatorWidget::update_display()
     else
         m_label->set_text("");
 }
+
+void CalculatorWidget::keydown_event(GKeyEvent& event)
+{
+    RefPtr<GButton> clicked_button;
```
Comment (bugaevc) [inline Applications/Calculator/CalculatorWidget.cpp]: Well it's certainly not clicked. Maybe it's `button_to_activate` or something like that.

**Comment by bugaevc** [comment]
What if instead of emulating clicks on `GButton`s, we perform the target action directly? Say, instead of finding a button with digit 2 and clicking it, you call `m_keypad.type_digit(2);` directly.

File: Applications/Calculator/CalculatorWidget.cpp (hunk 2 above)
Comment (bugaevc) [reply Applications/Calculator/CalculatorWidget.cpp]: (And no need to keep it in a `RefPtr` here, a raw pointer will do.)

**Comment by rhin123** [comment]
> What if instead of emulating clicks on `GButton`s, we perform the target action directly? Say, instead of finding a button with digit 2 and clicking it, you call `m_keypad.type_digit(2);` directly.

yeah that sounded better, done.. maybe 😁

**Comment by awesomekling** [comment]
Looks noice

**Pull request merged**